crates/driver/src/infra/solver/dto/auction.rs

```diff
@@ -223,7 +254,7 @@ struct ConstantProductPool {
     address: eth::H160,
     #[serde_as(as = "serialize::U256")]
     gas_estimate: eth::U256,
-    tokens: HashMap<eth::H160, ConstantProductReserve>,
+    tokens: BTreeMap<eth::H160, ConstantProductReserve>,
```

Comment:
Uniswap pool tokens are ordered (i.e. they have a concept of "token0 and token1" ordered by token address).

This changes the DTO to represent this.